README.md

```diff
@@ -113,6 +113,34 @@ However, if your project or any of its dependencies depend directly on compiler
 
 Because the internal APIs of the Kotlin compiler often change between versions, we can only support one `kotlin-compiler-embeddable` version at a time. 
 
+## Kotlin Symbol Processing API Support
+[Kotlin Symbol Processing (KSP)](https://goo.gle/ksp) is a new annotation processing pipeline that builds on top of the
+plugin architecture of the Kotlin Compiler, instead of delegating to javac as `kapt` does.
+
+**Note:** KSP is currently in active development and requires Kotlin 1.4 hence its support is kept separate from the
+main project until it becomes stable.
+
+To test KSP processors, you need to add a dependency to the ksp module:
+
+```Groovy
+dependencies {
+    implementation 'com.github.tschuchortdev:kotlin-compile-testing-ksp:1.2.9'
+}
+```
+
+This module adds a new function to the `KotlinCompilation` to specify KSP processors:
+
+```Kotlin
+class MySymbolProcessor : SymbolProcessor {
+    // implementation of the SymbolProcessor from the KSP API
+}
+
+val result = KotlinCompilation().apply {
+    sources = listOf(source)
+    symbolProcessors(MySymbolProcessor::class.java)
+}.compile()
+```
```

Comment:
Use of the KSP JUnit rule should be documented in the README or nobody will understand it. Maybe put those docs into another MD file and like to it here so that the README doesn't get too long.

Reply:
KSP JUnit rule is not public, it is just for my own KSP tests.
Getting the instance processors work is a bit tricky right now w/o being able to hook into compilation.
I'm planning to tackle it as a followup CL if you don't mind and then we wouldn't need this rule here either.

Reply:
Okay, if you don't find a better way to pass processor instance then it might be a good idea to make it public. It has issues with concurrency but so does the `ThreadLocal` solution for Kapt and it does make it much easier to write tests because you can write the test processor "inline" instead of creating a top-level processor class for every test case.